node/pkg/processor/observation.go

```diff
@@ -218,38 +204,60 @@ func (p *Processor) handleObservation(ctx context.Context, m *gossipv1.SignedObs
 			zap.Bool("quorum", len(sigs) >= quorum),
 		)
 
-		if len(sigs) >= quorum && !p.state.vaaSignatures[hash].submitted {
-			vaaBytes, err := signed.Marshal()
-			if err != nil {
-				panic(err)
+		if len(sigs) >= quorum && !p.state.signatures[hash].submitted {
+			switch o := p.state.signatures[hash].ourObservation.(type) {
+			case *vaa.VAA:
+				p.handleVAAQuorum(o, sigs, hash)
+			default:
+				panic("unknown observation type")
 			}
-
-			// Store signed VAA in database.
-			p.logger.Info("signed VAA with quorum",
-				zap.String("digest", hash),
-				zap.Any("vaa", signed),
-				zap.String("bytes", hex.EncodeToString(vaaBytes)),
-				zap.String("message_id", signed.MessageID()))
-
-			if err := p.db.StoreSignedVAA(signed); err != nil {
-				p.logger.Error("failed to store signed VAA", zap.Error(err))
-			}
-
-			p.broadcastSignedVAA(signed)
-			p.attestationEvents.ReportVAAQuorum(signed)
-			p.state.vaaSignatures[hash].submitted = true
 		} else {
 			p.logger.Info("quorum not met or already submitted, doing nothing",
 				zap.String("digest", hash))
 		}
 	} else {
-		p.logger.Info("we have not yet seen this VAA - temporarily storing signature",
+		p.logger.Info("we have not yet seen this observation - temporarily storing signature",
 			zap.String("digest", hash),
 			zap.Bools("aggregation", agg))
 
 	}
 }
 
+func (p *Processor) handleVAAQuorum(v *vaa.VAA, sigs []*vaa.Signature, hash string) {
+	// Deep copy the observation and add signatures
+	signed := &vaa.VAA{
+		Version:          v.Version,
+		GuardianSetIndex: v.GuardianSetIndex,
+		Signatures:       sigs,
+		Timestamp:        v.Timestamp,
+		Nonce:            v.Nonce,
+		Sequence:         v.Sequence,
+		EmitterChain:     v.EmitterChain,
+		EmitterAddress:   v.EmitterAddress,
+		Payload:          v.Payload,
+		ConsistencyLevel: v.ConsistencyLevel,
+	}
+	vaaBytes, err := signed.Marshal()
+	if err != nil {
+		panic(err)
```

Comment:
```suggestion
		panic(fmt.Errorf("problem unmarshalling signed VAA: %w", err))
```

Not necessarily better, but maybe?